src/pushsource/_impl/backend/registry_source.py

```diff
@@ -0,0 +1,171 @@
+import logging
+
+
+try:
+    from urllib.parse import urlparse
+except ImportError: # pragma: no cover
+    from urlparse import urlparse
+
+from ..source import Source
+from ..model import (
+    ContainerImagePushItem,
+    SourceContainerImagePushItem,
+    ContainerImagePullInfo,
+    ContainerImageTagPullSpec,
+    ContainerImageDigestPullSpec,
+)
+from ..utils.containers import (
+    get_manifest,
+    inspect,
+    MT_S2_V2,
+    MT_S2_V1,
+    MT_S2_V1_SIGNED,
+    MT_S2_LIST,
+)
+
+from ..helpers import list_argument
+
+LOG = logging.getLogger("pushsource")
+
+
+class RegistrySource(Source):
+    """Uses URIs of container images as source for push items."""
+
+    def __init__(
+        self,
+        dest=None,
+        registry_uris=None,
+        signing_key=None,
+        docker_url="unix://var/run/docker.sock",
+        executor_container_image=None,
+        docker_timeout=None,
+        docker_verify_tls=None,
+        docker_cert_path=None,
+    ):
+        """Create a new source.
+
+        Parameters:
+            dest (str, list[str])
+                The destination(s) to fill in for push items created by this
+                source. If omitted, all push items have empty destinations.
+
+            registry_uris (list[str])
+                List of references to container images with tags+dest tags
+                Format <host>/<namespace>/<repo>:<tag>:<destination_tag>:<destination_tag>
+                Example: registry.redhat.io/ubi:8:latest:8:8.1
+
+            signing_key (list[str])
+                GPG signing key ID(s). If provided, content must be signed
```

Comment:
This is not accurate, is it?  It looks like it was copy-pasted from koji backend, but in that backend it means "if we find the content and it's not signed with this key, it's a fatal error".  But here it instead just means "The value you pass will be copied onto the output push items".

Could it please be reworded for accuracy?

Also, I think it should be named dest_signing_key (matching exactly the related attr on the push items).  This makes the difference with KojiSource signing_key clearer, and also avoids a clash if we ever want to actually verify signing keys here similarly as the koji case.